libpod/runtime_ctr.go

```diff
@@ -144,25 +141,34 @@ func (r *Runtime) removeContainer(c *Container, force bool) error {
 		return err
 	}
 
-	if err := r.state.RemoveContainer(c); err != nil {
-		return errors.Wrapf(err, "error removing container from state")
+	// Remove the container from the state
+	if c.config.Pod != "" {
+		pod, err := r.state.Pod(c.config.Pod)
+		if err != nil {
+			return errors.Wrapf(err, "container %s is in pod %s, but pod cannot be retrieved", c.ID(), pod.ID())
+		}
+
+		if err := r.state.RemoveContainerFromPod(pod, c); err != nil {
+			return errors.Wrapf(err, "error removing container %s from state", c.ID())
+		}
+	} else {
+		if err := r.state.RemoveContainer(c); err != nil {
+			return errors.Wrapf(err, "error removing container from state")
+		}
 	}
 
 	// Delete the container
-	if err := r.ociRuntime.deleteContainer(c); err != nil {
-		return errors.Wrapf(err, "error removing container %s from runc", c.ID())
+	// Only do this if we're not ContainerStateConfigured - if we are,
+	// we haven't been created in runc yet
+	if c.state.State == ContainerStateConfigured {
+		if err := r.ociRuntime.deleteContainer(c); err != nil {
+			return errors.Wrapf(err, "error removing container %s from runc", c.ID())
```

Comment:
Should be oci runtime, we might support something other then runc at some point.